src/manifolds/Stiefel.jl

```diff
@@ -318,3 +318,26 @@ i.e. `(n,k)`, which is the matrix dimensions.
 @generated representation_size(::Stiefel{n,k}) where {n,k} = (n, k)
 
 show(io::IO, ::Stiefel{n,k,F}) where {n,k,F} = print(io, "Stiefel($(n), $(k), $(F))")
+
+"""
+    uniform_distribution(M::Stiefel, p)
+
+Uniform distribution on given [`Stiefel`](@ref) `M`. Generated points will be of
+similar type as `p`.
+"""
+function uniform_distribution(M::Stiefel, p)
+    d = RandnMatrix(size(p)...)
+    return ProjectedPointDistribution(M, d, project!, p)
+end
+
+struct RandnMatrix <: ContinuousMatrixDistribution
+    n::Integer
+    m::Integer
+end
```

Comment:
But we might ultimately want to switch over to `MatrixGaussian` when it is merged: https://github.com/JuliaStats/Distributions.jl/pull/1091. In a later PR, of course.

Reply:
I think also it's customary for distributions to store an eltype, so it probably be good to add that to the type.

Reply:
Generally agreed with your later suggestion to use `MatrixNormal` for now, but will commit this suggestion too so so that we also have the improved alternative handy. :)

Reply:
Agreed on both, I'm committing this suggestion and will add eltype in a follow-up commit.